--- flutter.js	2023-08-24 12:50:32
+++ flutter_new.js	2023-08-24 13:06:45
@@ -375,7 +375,19 @@
       // Install the `didCreateEngineInitializer` listener where Flutter web expects it to be.
       this.didCreateEngineInitializer =
         entrypointLoader.didCreateEngineInitializer.bind(entrypointLoader);
-      return entrypointLoader.loadEntrypoint(entrypoint);
+      const main = entrypointLoader.loadEntrypoint(entrypoint);
+
+      fetch("assets/FontManifest.json");
+      fetch("assets/fonts/MaterialIcons-Regular.otf");
+      fetch("assets/version.json");
+      fetch("assets/packages/feather_icons/fonts/feather.ttf");
+      fetch("assets/packages/font_awesome_flutter/lib/fonts/fa-brands-400.ttf");
+      fetch("assets/packages/font_awesome_flutter/lib/fonts/fa-regular-400.ttf");
+      fetch("assets/packages/font_awesome_flutter/lib/fonts/fa-solid-900.ttf");
+      fetch("assets/packages/line_icons/lib/assets/fonts/LineIcons.ttf");
+      fetch("assets/packages/material_design_icons_flutter/lib/fonts/materialdesignicons-webfont.ttf");
+
+      return main;
     }
   }
 
