--- flutter_service_worker.js	2023-08-23 11:36:34
+++ flutter_service_worker_new.js	2023-08-23 11:36:30
@@ -37,10 +37,7 @@
 "canvaskit/skwasm.worker.js": "51253d3321b11ddb8d73fa8aa87d3b15"};
 // The application shell files that are downloaded before a service worker can
 // start.
-const CORE = ["main.dart.js",
-"index.html",
-"assets/AssetManifest.json",
-"assets/FontManifest.json"];
+const CORE = ["main.dart.js"];
 
 // During install, the TEMP cache is populated with the application shell files.
 self.addEventListener("install", (event) => {
