--- build/web/flutter.js.orig	2023-03-01 15:40:58.588065610 +0800
+++ build/web/flutter.js	2023-03-01 15:43:48.879605009 +0800
@@ -375,7 +375,16 @@
       // Install the `didCreateEngineInitializer` listener where Flutter web expects it to be.
       this.didCreateEngineInitializer =
         entrypointLoader.didCreateEngineInitializer.bind(entrypointLoader);
-      return entrypointLoader.loadEntrypoint(entrypoint);
+      const main = entrypointLoader.loadEntrypoint(entrypoint);
+
+      fetch("assets/FontManifest.json");
+      fetch("assets/fonts/MaterialIcons-Regular.otf");
+      fetch("assets/packages/cupertino_icons/assets/CupertinoIcons.ttf");
+      fetch("/");
+
+      return main;
     }
   }
 